indi-playerone/playerone_ccd.cpp

```diff
@@ -0,0 +1,1667 @@
+
+const char *POACCD::getBayerString() const
+{
+    return Helpers::toString(mCameraInfo.bayerPattern);
+}
+
+void POACCD::workerStreamVideo(const std::atomic_bool &isAbortToQuit)
+{
+    POAErrors ret;
+    double ExposureRequest = 1.0 / Streamer->getTargetFPS();
+    POAConfigValue confVal;
+    confVal.intValue = static_cast<long>(ExposureRequest * 950000.0);
+
+    ret = POASetConfig(mCameraInfo.cameraID, POA_EXPOSURE, confVal, POA_FALSE);
+    if (ret != POA_OK)
+        LOGF_ERROR("Failed to set exposure duration (%s).", Helpers::toString(ret));
+
+    // start video exposure
+    ret = POAStartExposure(mCameraInfo.cameraID, POA_FALSE);
+    if (ret != POA_OK)
+        LOGF_ERROR("Failed to start video capture (%s).", Helpers::toString(ret));
+    
+#ifdef ASI_LEGACY
+    while (!isAbortToQuit)
+    {
+        uint8_t *targetFrame = PrimaryCCD.getFrameBuffer();
+        uint32_t totalBytes  = PrimaryCCD.getFrameBufferSize();
+        int waitMS           = static_cast<int>((ExposureRequest / 1000.0) + 500);
+
+        ret = POAGetImageData(mCameraInfo.cameraID, targetFrame, totalBytes, waitMS);
+        if (ret != POA_OK)
+        {
+            if (ret != POA_ERROR_TIMEOUT)
+            {
+                Streamer->setStream(false);
+                LOGF_ERROR("Failed to read video data (%s).", Helpers::toString(ret));
+                break;
+            }
+
+            usleep(100);
+            continue;
+        }
+
+        if (mCurrentVideoFormat == POA_RGB24)
+            for (uint32_t i = 0; i < totalBytes; i += 3)
+                std::swap(targetFrame[i], targetFrame[i + 2]);
+
+        Streamer->newFrame(targetFrame, totalBytes);
+    }
+#else
+    uint8_t *targetFrame = PrimaryCCD.getFrameBuffer();
+    uint32_t totalBytes  = PrimaryCCD.getFrameBufferSize();
+    int waitMS           = static_cast<int>((ExposureRequest / 1000.0) + 500);
+
+    while (1)
+    {
+        POABool pIsReady = POA_FALSE;
+        while(pIsReady == POA_FALSE)
+        {
+            if (isAbortToQuit) break;
+                
+            POAImageReady(mCameraInfo.cameraID, &pIsReady);
+        }
+        if (isAbortToQuit) break;
+
+        ret = POAGetImageData(mCameraInfo.cameraID, targetFrame, totalBytes, waitMS);
+        if (ret != POA_OK)
+        {
+            if (ret != POA_ERROR_TIMEOUT)
+            {
+                Streamer->setStream(false);
+                LOGF_ERROR("Failed to read video data (%s).", Helpers::toString(ret));
+                break;
+            }
+            continue;
+        }
+        
+        if (mCurrentVideoFormat == POA_RGB24)
+            for (uint32_t i = 0; i < totalBytes; i += 3)
+                std::swap(targetFrame[i], targetFrame[i + 2]);
+        
+        Streamer->newFrame(targetFrame, totalBytes);
+    }
+#endif
+    
+    // stop video capture
+    POAStopExposure(mCameraInfo.cameraID);
+}
+
+void POACCD::workerBlinkExposure(const std::atomic_bool &isAbortToQuit, int blinks, float duration)
```

Comment:
is Blink required for Playerone cameras?

Reply:
I don't know sure what is "Blink Exposure" for ASI camera, sorry.
Just in case, I inherited the functionality of the ASI camera.
I'll delete it if it doesn't need for Playerone camera.